docs/overview/architecture.md

```diff
@@ -0,0 +1,240 @@
+# Vald Architecture <!-- omit in toc -->
+
+## Table of Contents <!-- omit in toc -->
+
+- [Overview](#overview)
+- [Data Flow](#data-flow)
+  - [Insert](#insert)
+  - [Search](#search)
+- [Components](#components)
+  - [Vald Filter](#vald-filter)
+    - [Vald Ingress Filter](#vald-ingress-filter)
+    - [Vald Egress Filter](#vald-egress-filter)
+    - [Vald Filter Gateway](#vald-filter-gateway)
+  - [Vald Metadata](#vald-metadata)
+    - [Vald Meta Gateway](#vald-meta-gateway)
+    - [Vald Meta](#vald-meta)
+  - [Vald Backup](#vald-backup)
+    - [Vald Compressor](#vald-compressor)
+    - [Vald Backup Manager](#vald-backup-manager)
+    - [Vald Backup Gateway](#vald-backup-gateway)
+  - [Vald Load Balancing](#vald-load-balancing)
+    - [Vald LB Gateway](#vald-lb-gateway)
+    - [Agent Discoverer](#agent-discoverer)
+  - [Vald Core Engine](#vald-core-engine)
+    - [Vald Agent](#vald-agent)
+    - [Vald Agent Scheduler](#vald-agent-scheduler)
+    - [Vald Index Manager](#vald-index-manager)
+  - [Vald Replication Manager](#vald-replication-manager)
+    - [Vald Replication Manager Agent](#vald-replication-manager-agent)
+    - [Vald Replication Manager Controller](#vald-replication-manager-controller)
+  - [Kubernetes Components](#kubernetes-components)
+    - [Kube-apiserver](#kube-apiserver)
+    - [Custom Resources](#custom-resources)
+
+## Overview
+
+This document describes the high-level architecture design of Vald and explains each component in Vald.
+We need these components to support scalability, high performance, and auto-healing in Vald.
```

Comment:
I'm not sure, this sentence is suited here 😅

Reply:
If we remove the sentence before, it is not suit here..

Reply:
yes.
Do you have any ideas?

Reply:
I am thinking it is better to remove the sentence, what do you think?